### PR TITLE
fix boot_cpu_model.check_flag.remove_flag/cpu_host on Opteron_G3-G5

### PR DESCRIPTION
**Pull request by XueqiangWei**
Signed-off-by: Xueqiang Wei xuwei@redhat.com

### CODE REVIEW

**Comment by XueqiangWei** [comment]
ID 1265129
ID 1265105

**Comment by xutian** [comment]
@XueqiangWei  please add commit message to description what issue exists and how you fixed it.

Thanks,
Xu

File: qemu/deps/cpuid/cpuid_dump.txt [hunk 1]
```diff
@@ -1,20 +1,20 @@
 x86       Opteron_G5 AMD Opteron 63xx class CPU
   family 21 model 2 stepping 0 level 13 xlevel 0x80000001a vendor "AuthenticAMD"
-  feature_edx (sse2 sse fxsr mmx clflush pse36 pat cmov mca pge mtrr sep apic cx8 mce pae msr tsc pse de fpu)
-  feature_ecx (f16c avx xsave aes popcnt sse4.2|sse4_2 sse4.1|sse4_1 cx16 pma ssse3 pclmulqdq|pclmuldq pni|sse3)
+  feature_edx (vme sse2 sse fxsr mmx clflush pse36 pat cmov mca pge mtrr sep apic cx8 mce pae msr tsc pse de fpu)
```
Comment (suqinhuang) [inline qemu/deps/cpuid/cpuid_dump.txt]: could you share me where is flag "vme" defined for Opteron_G5

File: qemu/deps/cpuid/cpuid_dump.txt [hunk 2]
```diff
@@ -43,7 +43,7 @@ x86         Haswell Intel Core Processor (Haswell)
 
 x86         SandyBridge Intel Xeon E312xx (Sandy Bridge)
   family 6 model 42 stepping 1 level 13 xlevel 0x8000000a vendor "GenuineIntel"
-  feature_edx (sse2 sse fxsr mmx clflush pse36 pat cmov mca pge mtrr apic cx8 mce pae msr tsc pse de fpu)
+  feature_edx (sse2 sse fxsr mmx clflush pse36 pat cmov mca pge mtrr sep apic cx8 mce pae msr tsc pse de fpu)
```
Comment (suqinhuang) [inline qemu/deps/cpuid/cpuid_dump.txt]: SEP should be supported in all cpu module, but need to make sure -M > rhel6.4

File: qemu/tests/cfg/boot_cpu_model.cfg [hunk 3]
```diff
@@ -49,9 +49,16 @@
...
+                    cpu_model_flags_AuthenticAMD = ",+cmp_legacy"
                 - cpuid_7_0_ebx_feature:
                     cpu_model_flags = ",+fsgsbase"
                     cpu_model = "SandyBridge"
                 - cpu_host:
                     cpu_model = "host"
+                    cpu_model_flags = 
+                    AuthenticAMD_option_flags = " tsc_deadline_timer"
+                    Opteron_G3_extra_flags = ",-lahf_lm,+3dnow,+pdpe1gb,+3dnowprefetch,+3dnowext"
+                    Opteron_G4_extra_flags = ",-x2apic"
```
Comment (suqinhuang) [inline qemu/tests/cfg/boot_cpu_model.cfg]: if flag is shown in guest, but no need to check it, pls set in option_flags

File: qemu/tests/cfg/boot_cpu_model.cfg [hunk 4]
```diff
@@ -49,9 +49,16 @@
...
+                    cpu_model_flags = ",-sse4a,-xsave,-aes,-sse4.2,+ht"
+                    cpu_model_flags_AuthenticAMD = ",+cmp_legacy"
                 - cpuid_7_0_ebx_feature:
                     cpu_model_flags = ",+fsgsbase"
                     cpu_model = "SandyBridge"
                 - cpu_host:
                     cpu_model = "host"
+                    cpu_model_flags = 
+                    AuthenticAMD_option_flags = " tsc_deadline_timer"
+                    Opteron_G3_extra_flags = ",-lahf_lm,+3dnow,+pdpe1gb,+3dnowprefetch,+3dnowext"
```
Comment (suqinhuang) [inline qemu/tests/cfg/boot_cpu_model.cfg]: Confused why need to add "+3dnow,+pdpe1gb,+3dnowprefetch,+3dnowext"

File: qemu/tests/cfg/boot_cpu_model.cfg [hunk 5]
```diff
@@ -49,9 +49,16 @@
                 - remove_flag:
                     cpu_model_GenuineIntel = "SandyBridge"
                     cpu_model_AuthenticAMD = "Opteron_G3"
-                    cpu_model_flags = ",-sse4a,-xsave,-aes,-sse4.2"
+                    cpu_model_flags = ",-sse4a,-xsave,-aes,-sse4.2,+ht"
+                    cpu_model_flags_AuthenticAMD = ",+cmp_legacy"
```
Comment (suqinhuang) [inline qemu/tests/cfg/boot_cpu_model.cfg]: Did you get the reason why there is cmp_legacy, but cmplegacy is definded in both host and guest

File: qemu/deps/cpuid/cpuid_dump.txt (hunk 1 above)
Comment (XueqiangWei) [inline qemu/deps/cpuid/cpuid_dump.txt]: some texts from cpu.c in blew: 
.name = "Opteron_G5",
        ......       
       .features[FEAT_1_EDX] =
            CPUID_VME | CPUID_SSE2 | CPUID_SSE | CPUID_FXSR | CPUID_MMX |
            .......

File: qemu/deps/cpuid/cpuid_dump.txt (hunk 2 above)
Comment (XueqiangWei) [inline qemu/deps/cpuid/cpuid_dump.txt]: I see, in the avocado code, the defaule value is  -M > rhel6.4, so I add sep to dumpfile. Also we can use +sep fix this issue.

File: qemu/tests/cfg/boot_cpu_model.cfg (hunk 3 above)
Comment (XueqiangWei) [inline qemu/tests/cfg/boot_cpu_model.cfg]: tsc_deadline_timer is Intel-defined CPU features, not in AMD host and dumpfile, but it is in Guest. So set in AMD_option_flags.

File: qemu/tests/cfg/boot_cpu_model.cfg (hunk 4 above)
Comment (XueqiangWei) [inline qemu/tests/cfg/boot_cpu_model.cfg]: these flags are in host but not in dumpfile, need to expose to guest.

File: qemu/tests/cfg/boot_cpu_model.cfg (hunk 5 above)
Comment (XueqiangWei) [inline qemu/tests/cfg/boot_cpu_model.cfg]: in file cpufeature.h, this feature is defined as below
#define X86_FEATURE_CMP_LEGACY
in the qemu: it is defined as cmplegacy
Although the names are different, they all correspond to a cpu register.

**Comment by xutian** [comment]
Please split your commit to small fix, view from change I don't know what you want to do.

Thanks,
Xu

**Comment by XueqiangWei** [comment]
wait for developper's feedback, and then update this pull request.

**Comment by maxujun** [comment]
smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u2.spice.qcow2.virtio_scsi.up.virtio_net.RHEL.6.8.x86_64.io-github-autotest-qemu.boot_cpu_model.check_flag.remove_flag: -** Message: pygobject_register_sinkfunc is deprecated (GstObject)
PASS
smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u2.spice.qcow2.virtio_scsi.up.virtio_net.RHEL.6.8.x86_64.io-github-autotest-qemu.boot_cpu_model.check_flag.cpu_host: -** Message: pygobject_register_sinkfunc is deprecated (GstObject)
PASS
LGTM
Acked-by:xuma@redhat.com

**Comment by xutian** [comment]
@suqinhuang  can you review this PR?

**Comment by XueqiangWei** [comment]
the test has moved to avocado, will resend a new pullreq.

**Pull request closed without merging**